ambry-rest/src/main/java/com/github/ambry/rest/NettyResponseChannel.java

```diff
@@ -681,10 +706,17 @@ private HttpResponseStatus getHttpResponseStatus(ResponseStatus responseStatus)
    * </p>
    * May also close the channel if the class internally is forcing a close (i.e. if {@link #close()} is called.
    * @param closeNetworkChannel network channel is closed if {@code true}.
+   * @param shouldDelay network channel may be closed after certain delay if {@code true}.
    */
-  private void completeRequest(boolean closeNetworkChannel) {
+  private void completeRequest(boolean closeNetworkChannel, boolean shouldDelay) {
     if ((closeNetworkChannel || forceClose) && ctx.channel().isOpen()) {
-      writeFuture.addListener(ChannelFutureListener.CLOSE);
+      if (shouldDelay && (request != null && request.getRestMethod().equals(RestMethod.POST))
```

Comment:
I wonder if we need to only do this on post requests? It is probably less likely to experience this issue on non-POST requests since those don't usually have a body and closeNetworkChannel is rarely true, but I don't see an issue with doing the delayed close on all types of requests where closeNetworkChannel is true

Reply:
Thanks @cgtz for the comment. Later I was thinking about doing this for all http method as well but since the ticket specifically talks about to solve it for POST method I made it like so. Please let me know if this fine or should i remove the request method check?

Reply:
Thanks @cgtz for the comment. Later I was thinking about doing this for all http method as well but since the ticket specifically talks about to solve it for POST method I made it like so. Please let me know if this fine or should i remove the request method check?

Reply:
I guess it is fine to keep it as is, since this issue probably happens when there is more request data to send, which is only posts for now. We can revisit this if we support other operations that contain request bodies in the future.

Reply:
Thanks for the confirmation. So will keep it as it is for now.